### PR TITLE
[OSD-19300] allow STS based access to new AWS accounts

### PR DESCRIPTION
**Pull request by reedcort**
# What is being added?
_Is this a fix for a bug?  What's the bug?  Is this a new feature? Please describe it. Is this just a small typo fix?  That's fine too!_

## Checklist before requesting review

- [x] I have tested this locally
- [ ] I have included unit tests
- [ ] I have updated any corresponding documentation

## Steps To Manually Test
_Please provide us steps to reproduce the scenarios needed to test this.  If an integration test is provided, let us know how to run it. If not, enumerate the steps to validate this does what it's supposed to._
1. Start the operator
1. Run the thing
1. Observe X in the spec for the thing
1. Clean up the thing

Ref OSD-0000

### CODE REVIEW

File: controllers/accountclaim/accountclaim_controller.go [hunk 1]
```diff
@@ -2,12 +2,17 @@ package accountclaim
...
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/iam"
+	stsclient "github.com/openshift/aws-account-operator/pkg/awsclient/sts"
+	"strings"
 	"time"
 
 	"github.com/go-logr/logr"
 	"github.com/openshift/aws-account-operator/config"
-	"github.com/openshift/aws-account-operator/controllers/account"
+	account "github.com/openshift/aws-account-operator/controllers/account"
```
Comment (AlexVulaj) [inline controllers/accountclaim/accountclaim_controller.go:15]: any reason for adding the alias here?

File: controllers/accountclaim/accountclaim_controller.go [hunk 2]
```diff
@@ -231,6 +386,176 @@ func (r *AccountClaimReconciler) Reconcile(ctx context.Context, request ctrl.Req
...
+	// Attach the permissions policy to the role
+	_, err = awsClient.PutRolePolicy(&iam.PutRolePolicyInput{
+		PolicyName:     aws.String(stsPolicyName),
+		RoleName:       aws.String(roleName),
+		PolicyDocument: aws.String(policyDocument),
+	})
+
+	if err != nil {
+		// If there was an error, clean up by deleting the role
+		awsClient.DeleteRole(&iam.DeleteRoleInput{
```
Comment (AlexVulaj) [inline controllers/accountclaim/accountclaim_controller.go:551]: Can this return an error? If so, let's handle (or at least log) it here.

File: controllers/accountclaim/accountclaim_controller.go [hunk 3]
```diff
@@ -231,6 +393,181 @@ func (r *AccountClaimReconciler) Reconcile(ctx context.Context, request ctrl.Req
...
+		AssumeRolePolicyDocument: aws.String(string(jsonAssumeRolePolicyDoc)),
+	})
+	if err != nil {
+		return "", err
+	}
+	reqLogger.Info(fmt.Sprintf("Role %s created", createRoleOutput))
+
+	arnComponents := strings.Split(*createRoleOutput.Role.Arn, ":")
+
+	policyDocument, err := generateInlinePolicy(arnComponents[4])
```
Comment (bergmannf) [inline controllers/accountclaim/accountclaim_controller.go:546]: I think [4] is the `account`, so maybe assign it to a variable named accountid, to make it more clear.

File: api/v1alpha1/accountclaim_types.go [hunk 4]
```diff
@@ -13,21 +13,22 @@ import (
...
+	BYOCSecretRef       SecretRef          `json:"byocSecretRef,omitempty"`
+	BYOCAWSAccountID    string             `json:"byocAWSAccountID,omitempty"`
+	ManualSTSMode       bool               `json:"manualSTSMode,omitempty"`
+	STSRoleARN          string             `json:"stsRoleARN,omitempty"`
+	STSExternalID       string             `json:"stsExternalID,omitempty"`
+	SupportRoleARN      string             `json:"supportRoleARN,omitempty"`
+	CustomTags          string             `json:"customTags,omitempty"`
+	KmsKeyId            string             `json:"kmsKeyId,omitempty"`
+	AccountPool         string             `json:"accountPool,omitempty"`
+	FleetManagerConfig  FleetManagerConfig `json:"fleetManagerConfig,omitempty"`
```
[the remark this answers is not in the document]
Comment (reedcort) [reply api/v1alpha1/accountclaim_types.go:31]: Ok I'll add comment mentioning "FleetmanagerConfig is exclusively designed for use by the fleet manager" and I'll cut a card to update the documentation and explain this in more detail.

**Pull request merged**